### ISSUE TITLE
is it possible to use the XML parser for a different project?

### ISSUE THREAD

**Issue by martinlippert** [angelozerr/lsp4xml#280]
I am interested in re-using the XML parser implementation for a different project (ideally as a maven dependency). Do you think that would be possible?

I am especially interested in a simple XML parser (could be DOM or StAX based) that delivers exact position information for the parsed elements and this seems like something your parser does.

Or are you aware of another XML parser lib that does this?

Thanks a lot for your help!!!

**Comment by angelozerr** [comment]
> I am especially interested in a simple XML parser (could be DOM or StAX based) that delivers exact position information for the parsed elements and this seems like something your parser does.

Exactly!

The main difference between our own DOMParsre and SAX parser is that:

 * it is tolerant
 * it gives you node location for each nodes (element, attributes, text, etc)
 * it doesn't manage update (just read).

Our DOMParsre implements w3c DOM structures but it doesn't manage updat e(like insertNode, etc)

I think you could use DOMParser in your own java main (lsp4j library is required). See simple load of DOM at https://github.com/angelozerr/lsp4xml/blob/master/org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/dom/DOMDocumentTest.java#L39

**Comment by martinlippert** [comment]
Sounds promising. Do you have the library published as a maven artifact somewhere on maven central?

**Comment by martinlippert** [comment]
@fbricon sounds promising... :-) Will you update this issue then? That would be awesome!!!

**Comment by fbricon** [comment]
lsp4xml 0.3.0 has been released, available as a maven artifact, see #229.

**Comment by martinlippert** [comment]
AWESOME !!!!! Many many thanks for the update and the published artifact, already using it for the first increment of parsing some Spring symbols out of XML config files... :-)

**Comment by angelozerr** [comment]
It's a great news @martinlippert ! Please give us your feedback, is it fast, have you found some bugs, etc.

Could you create a PR please to add in the README that you are using our XML DOM parser with a link to your project. Thanks!

**Comment by martinlippert** [comment]
@angelozerr I created the pull request for the README and wanted to say a super big THANK YOU for this work. We are not re-using the language server as-is, but only specific parts of this (especially the fault-tolerant XML parser and scanner) to implement Spring-Tools-specific analysis of XML files and it is a real pleasure to work with this lib. Awesome! Many many thanks!!!

We also started to re-use this lib for early versions of Spring-specific content-assist in XML files. We decided against re-using the full-blown language server with specific extensions and instead re-using specific pieces as a library. The reason is to allow people to continue to use their existing XML editor/tooling (either the old-fashioned XML editor in Eclipse, the generic one in Eclipse, or the VSCode extension) and then let our Spring language server add only specific features to that as a separate language server.
